tools/scoring/score/parse_config.py

```diff
@@ -177,3 +178,57 @@ def retrieve_reporting_translations(
       }
 
     return translations
+
+  @staticmethod
+  def aggregate_results_nondbo(
+      # TODO: create Dimension type to replace generic
+      *,
+      dimensions: List[Callable[[Dict], Dict]],
+      translations: Dict[str, List[Tuple[str, Any]]]) -> Dict[str, Result]:
+    """
+      Wrapper which outputs a dictionary of results by invoking
+      each specified `Dimension` with the `translations` argument
+
+      Args:
+        dimensions: List of `Dimension`s to be evaluated
+        translations: Dictionary with `cloud_device_id`s as keys
+          and lists of translation tuples as values
+
+      Returns:
+        Dictionary with dimension names as keys and `Result`s as values
+    """
+    results = {}
+    for dimension in dimensions:
+      # Invoke the function and append the dictionary with its return value
+      results[dimension.__name__] = dimension(translations=translations)
+    return results
+
+  @staticmethod
+  def aggregate_results_dbo(
```

Comment:
aggregate_results_by_dimension?
Are we calculating here the result by Dimension?

What is a dimension?
reporting devices
field matching etc all those are a dimension each?

it is a sort of a measurement?

Reply:
The suffix is for differentiating between the two types of dimension: less complex (which operate on the translations of each matched reporting pair) and more (which operate on the entire files). The categorization is almost meaningless save for the common signatures. You figured it out: a dimension is a measurement, and these methods just wrap every dimension for modularity.